src/services/Markdown.ts

```diff
@@ -0,0 +1,44 @@
+import fm from 'front-matter';
+
+import { MarkdownDocument } from '@/types/MarkdownDocument';
+import { fetchWithTimeout } from '@/utils/fetchWithTimeout';
+
+export default class Markdown {
+  async fetchMarkdown(file: string): Promise<MarkdownDocument> {
+    try {
+      const timeoutMs = 4000;
+      const response = await fetchWithTimeout(file, this.getMarkdownFetchInit(), timeoutMs);
+      if (!response.ok) {
```

Comment:
Figure out if we want to handle this by throwing an error or what we should do in this case